src/AppBridgeBlock.ts

```diff
@@ -289,6 +289,8 @@ export class AppBridgeBlock {
         if (!result.success) {
             throw new Error('Could not update the block settings');
         }
+
+        window.blockSettings[this.blockId] = newSettings as Record<string, unknown>;
```

Comment:
```suggestion
        window.blockSettings[this.blockId] = mergeDeep(window.blockSettings[this.blockId] || {}, newSettings);
```
So it is the same as the API and this is what we do in `useBlockSettings`